### PR TITLE
Add legacy and new SBOM support, with tests

### LINKED ISSUE #7: Add dependency SBOM
**Given** A developer is using this buildpack to install `composer`
**When** The developer inspects the launch image created by a lifecycle
**Then** The developer should see an SBOM for the dependency downloaded by this buildpack (`composer`)

## Notes:
- Use `paketo-buildpacks/packit/v2` `"sbom".GenerateFromDependency` to ensure that this SBOM has the right information

### PR DESCRIPTION
**Pull request by joshuatcasey**
## Summary
Add legacy and new SBOM support, with tests. Resolves #7.

## Checklist
* [x] I have viewed, signed, and submitted the Contributor License Agreement.
* [x] I have linked issue(s) that this PR should close using keywords or the Github UI (See [docs](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue))
* [x] I have added an integration test, if necessary.
* [x] I have reviewed the [styleguide](https://github.com/paketo-buildpacks/community/blob/main/STYLEGUIDE.md) for guidance on my code quality.
* [x] I'm happy with the commit history on this PR (I have rebased/squashed as needed).

### CODE REVIEW

**Comment by sophiewigmore** [comment]
@joshuatcasey The only thing I am thinking about is whether we should roll out the newer SBOM in the first release of the Composer CNB. For the rewrite, we only need to support the legacy. I wonder if it's confusing to also support the newer SBOM functionality in the first release.

**Comment by joshuatcasey** [comment]
@sophiewigmore Ok I can combine the integration tests. 

Do we know what might happen if I include those CPEs? They aren't actually listed here: https://nvd.nist.gov/products/cpe, although earlier versions of Composer are listed.

**Comment by sophiewigmore** [comment]
OK that's probably why the CPE wasn't included in the dep-server. The CPE will not be over written by our automation until a new version comes out I think so it should be OK. My main concern is making sure the CPE you added is accurate.

File: build_test.go [hunk 1]
```diff
@@ -118,6 +127,17 @@ func testBuild(t *testing.T, context spec.G, it spec.S) {
 					Metadata: map[string]interface{}{
 						"dependency-sha": "some-sha",
 					},
+					SBOM: sbom.Formatter{},
```
Comment (sophiewigmore) [inline build_test.go:130]: I'd love to see one test here where the `SBOMFormats` in the build context are set, and then we can see that the build result layer has some context like we do here: https://github.com/paketo-buildpacks/node-engine/blob/71ab3af64a0dd37947f4329381209723878c7773/build_test.go#L141-L150

**Pull request merged**